keps/693-multikueue/README.md

```diff
@@ -0,0 +1,240 @@
+    Spec MultiKueueConfigSpec `json:"spec,omitempty"`
+}
+
+type MultiKueueConfigSpec struct {
+     // List of MultiKueueWorkerClusters names where the 
+     // workloads from the ClusterQueue should be distributed.
+     Clusters []string `json:"clusters,omitempty"`
+}
+
+type MultiKueueCluster struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec MultiKueueClusterSpec `json:"spec,omitempty"`
+    Status MultiKueueClusterStatus `json:"status,omitempty"`
+}
+
+const (
+    // Location is the path on the disk.
+    PathLocationType LocationType = "Path"
+    
+    // Location is the name of the secret inside kueue-system namespace.
+    SecretLocationType LocationType = "Secret"
+)
+
+type MultiKueueClusterSpec {
+    // Information how to connect to the cluster.
+    KubeConfig KubeConfig `json:"kubeConfig"`
+}
+
+type KubeConfig struct {
+    // Name of the cluster inside the given KubeConfig.
+    Name string `json:"configName"`
+    
+    // Location of the KubeConfig.
+    Location string `json:"location"`
+    
+    // Type of the KubeConfig location.
+    LocationType LocationType `json:"locationType"`
+}
+
+type MultiKueueClusterStatus {
+   Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+```
+
+MultiKueue controller will monitor all cluster definitions and maintain 
+the Kube clients for all of them. Any connectivity problems will be reported both in
+MultiKueuCluster status as well as AdmissionCheckStatus and Events. MultiKueue controller 
+will make sure that whenever the kubeconfig is refreshed, the appropriate 
+clients will also be recreated.
+
+Creation of kubeconfig files is outside of the MultiKueue scope, and is cloud
+provider/environment dependant.
+
+MultiKueue controller, when pushing workloads to the worker clusters, will use the same 
+namespace and local queue names as were used in the management cluster. It is user's 
+responsibility to set up the appropriate namespaces and local queues.
+Worker ClusterQueue definitions may be different than in the management cluster. For example,
+quota settings may be specific to the given location. And/or cluster queue may have different
+admission checks, use ProvisioningRequest, etc.
+
+When distributing the workloads across clusters MultiKueue controller will first create
+the Kueue's internal Workload object. Only after the workload is admitted and other clusters
+are cleaned-up the real job will be created, to match the Workload. That gives the guarantee
+that the workload will not start in more than one cluster.
+
+When the job is running MultiKueue controller will copy its status from worker cluster
+to the management cluster, to keep the impression that the job is running in the management 
+cluster. This is needed to allow pipelines and workflow engines to execute against 
+the management cluster with MultiKueue without any extra changes. 
```

Comment:
Should we have an annotation for the name of the cluster where the job/workload is actually running?

Reply:
Added.